crates/circuit/src/dag_circuit.rs

```diff
@@ -2593,30 +2592,65 @@ def _format(operand):
         Ok(tup.into_any().iter().unwrap().unbind())
     }
 
-    /// Iterator for edge values and source and dest node
+    /// Iterator for edge values with source and destination node.
     ///
-    /// This works by returning the output edges from the specified nodes. If
+    /// This works by returning the outgoing edges from the specified nodes. If
     /// no nodes are specified all edges from the graph are returned.
     ///
     /// Args:
-    ///     nodes(DAGOpNode, DAGInNode, or DAGOutNode|list(DAGOpNode, DAGInNode, or DAGOutNode):
+    ///     nodes(DAGOpNode, DAGInNode, DAGOutNode or Vec of DAGOpNode, DAGInNode, or DAGOutNode:
     ///         Either a list of nodes or a single input node. If none is specified,
     ///         all edges are returned from the graph.
     ///
     /// Yield:
-    ///     edge: the edge in the same format as out_edges the tuple
-    ///         (source node, destination node, edge data)
-    fn edges(&self, nodes: Option<Bound<PyAny>>) -> Py<PyIterator> {
-        // if nodes is None:
-        //     nodes = self._multi_graph.nodes()
-        //
-        // elif isinstance(nodes, (DAGOpNode, DAGInNode, DAGOutNode)):
-        //     nodes = [nodes]
-        // for node in nodes:
-        //     raw_nodes = self._multi_graph.out_edges(node._node_id)
-        //     for source, dest, edge in raw_nodes:
-        //         yield (self._multi_graph[source], self._multi_graph[dest], edge)
-        todo!()
+    ///     edge: the edge as a tuple with the format
+    ///         (source node, destination node, edge wire)
+    fn edges(&self, nodes: Option<Bound<PyAny>>, py: Python) -> PyResult<Py<PyIterator>> {
+
+        let get_node_index = |obj: Bound<PyAny>| -> PyResult<NodeIndex> {
+            Ok(obj.downcast::<DAGNode>()?.borrow().node.unwrap())
+        };
+
+        let actual_nodes: Vec<_> = match nodes {
+            None => self.dag.node_references().map(|(index, weight)| index).collect(),
+            Some(nodes) => {
+                if let Ok(node) = get_node_index(nodes.clone()) {
+                    let mut out = Vec::new();
+                    out.push(node);
+                    out
+                } else {
+                    let mut out = Vec::new();
+                    for node in nodes.iter()?
+                    {
+                        out.push(get_node_index(node?)?);
+                    }
+                     out
+                 }
+            }
+        };
+
+        let mut edges = Vec::new();
+        for node in actual_nodes {
+            let in_out_nodes = self.dag.edges_directed(node, Outgoing).map(|e| {
```

Comment:
It might make sense to combine this `map` with the `for` loop below to avoid iterating through twice. And, if you're not using the `map` you can use the `?` directly on your calls to `get_node` and any errors will propagate up throw the `edges` method.